frontend/src/screens/achievements.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useState, useEffect } from 'react';
+import { Box, Text, Button, Container, Title, Tabs, RingProgress, Image, Group, Stack, Flex, Modal, Overlay, Alert } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { useApi } from '@lib';
+import { launchStudy } from '@models';
+import { TopNavBar, Footer } from '@components';
+import { colors } from '@theme';
+import { useFetchLearningPaths } from './learner/studies';
+import { ParticipantStudy } from '@api';
+import { useCurrentUser } from '@lib';
+
+const AchievementBadge:FC<{learningPath: string; studies: ParticipantStudy[]}> = ({ learningPath, studies }) => {
+
+    interface Criteria{
+        title: string,
+        text: string
+    }
+
+    const api = useApi()
+    const user = useCurrentUser()
+    const [progress, setProgress] = useState(0)
+    const [isComplete, setIsComplete] = useState(false)
+    const [criteria, setCriteria] = useState<Criteria[]>([] || undefined)
+    const [completedStudies, setCompletedStudies] = useState<ParticipantStudy[]>([])
+    const [btnText, setBtnText] = useState('Start')
+    const [error, setError] = useState(false)
+    const [opened, { open, close }] = useDisclosure(false)
+
+    useEffect(() => {
+        setCompletedStudies(studies.filter((study: ParticipantStudy) => study.completedAt))
+        setCriteria(() => {
+            return studies[0].learningPath?.badge?.criteriaHtml?.split('**')
+                .slice(1)
+                .reduce<Criteria[]>((acc: Criteria[], item, index, array) => {
+                if (index % 2 === 0) {
+                    acc.push({ title: item.trim(), text: array[index + 1].trim() });
+                }
+                return acc;
+            }, []) || []; 
+        })
+        
+    }, [studies])
+
+    useEffect(() => {
+        setProgress((completedStudies.length / studies.length) * 100)
+        setIsComplete(completedStudies.length == studies.length)
+    }, [completedStudies, isComplete])
+
+    useEffect(() => {
+        setBtnText(() => {
+            if(isComplete){
```

Comment:
could you just use these `isComplete` and `completedStudies.length` in the template to render the text? do we need to keep this in state?

Reply:
I can use isComplete and completedStudies.length